test/integrations/destinations/campaign_manager/dataDelivery/other.ts

```diff
@@ -0,0 +1,533 @@
+    module: 'destination',
+    version: 'v0',
+    input: {
+      request: {
+        body: generateProxyV0Payload({
+          endpoint: 'https://random_test_url/test_for_null_and_no_status',
+        }),
+        method: 'POST',
+      },
+    },
+    output: {
+      response: {
+        status: 500,
+        body: {
+          output: {
+            status: 500,
+            message:
+              'Campaign Manager: undefined during CAMPAIGN_MANAGER response transformation 3',
+            destinationResponse: {
+              response: '',
+              status: 500,
+            },
+            statTags: {
+              errorCategory: 'network',
+              errorType: 'retryable',
+              destType: 'CAMPAIGN_MANAGER',
+              module: 'destination',
+              implementation: 'native',
+              feature: 'dataDelivery',
+              destinationId: 'default-destinationId',
+              workspaceId: 'default-workspaceId',
+            },
+          },
+        },
+      },
+    },
+  },
+];
+
+export const otherScenariosV1 = [
+  {
+    id: 'cm360_v1_other_scenario_1',
+    name: 'campaign_manager',
+    description:
+      '[Proxy v1 API] :: Scenario for testing Service Unavailable error from destination',
+    successCriteria: 'Should return 500 status code with error message',
+    scenario: 'Framework',
+    feature: 'dataDelivery',
+    module: 'destination',
+    version: 'v1',
+    input: {
+      request: {
+        body: generateProxyV1Payload({
+          endpoint: 'https://random_test_url/test_for_service_not_available',
+        }),
+        method: 'POST',
+      },
+    },
+    output: {
+      response: {
+        status: 500,
+        body: {
+          output: {
+            response: [
+              {
+                error:
+                  '{"error":{"message":"Service Unavailable","description":"The server is currently unable to handle the request due to temporary overloading or maintenance of the server. Please try again later."}}',
+                statusCode: 500,
+                metadata: {
+                  jobId: 1,
+                  attemptNum: 1,
+                  userId: 'default-userId',
+                  destinationId: 'default-destinationId',
+                  workspaceId: 'default-workspaceId',
+                  sourceId: 'default-sourceId',
+                  secret: {
+                    accessToken: 'default-accessToken',
+                  },
+                  dontBatch: false,
+                },
+              },
+            ],
+            statTags: {
+              errorCategory: 'network',
+              errorType: 'retryable',
+              destType: 'CAMPAIGN_MANAGER',
+              module: 'destination',
+              implementation: 'native',
+              feature: 'dataDelivery',
+              destinationId: 'default-destinationId',
+              workspaceId: 'default-workspaceId',
+            },
+            authErrorCategory: '',
```

Comment:
`authErrorCategory` is missing in v0 but present in v1.

Reply:
Its because in `v0` error handler if auth error category is non empty then we are adding it but in case of `v1` its being added irrespective
v0 --> https://github.com/rudderlabs/rudder-transformer/blob/686f5246d8b90a45e85d451d1c2ae47b2512a190/src/services/destination/postTransformation.ts#L155

v1 -->https://github.com/rudderlabs/rudder-transformer/blob/686f5246d8b90a45e85d451d1c2ae47b2512a190/src/services/destination/postTransformation.ts#L189

Reply:
Yes addressed [here](https://github.com/rudderlabs/rudder-transformer/pull/3049)